apps/src/StudioApp.js

```diff
@@ -3154,6 +3154,25 @@ StudioApp.prototype.isResponsiveFromConfig = function(config) {
   return config.embed || isWorkspaceView;
 };
 
+/**
+ * Checks if the level a teacher is viewing of a students has
+ * not been started.
+ */
+StudioApp.prototype.isNotStartedLevel = function(config) {
+  const progress = getStore().getState().progress;
+
+  if (
+    ['Gamelab', 'Applab', 'Weblab', 'Spritelab'].includes(config.levelGameName)
+  ) {
+    return config.readonlyWorkspace && !config.channel;
+  } else if (!config.level.freePlay) {
+    return (
+      config.readonlyWorkspace &&
+      progress.levelProgress[progress.currentLevelId] === undefined
+    );
```

Comment:
What does this do?

Reply:
This is just another condition when we know that a level has not been started yet by a student. In levels that are not free play if a student has not ever run the program the levelProgress will be undefined.